plugins/libplugin.h

```diff
@@ -219,6 +219,10 @@ struct command_result *forward_result(struct command *cmd,
  * send_req() path. */
 struct command_result *timer_complete(struct plugin *p);
 
+/* Signals that we've completed a command. Useful for when
+ * there's no `cmd` present */
+struct command_result *command_done(void);
+
```

Comment:
Drop this commit.  The one place you end up using it, you do indeed have a command.  You just don't use it!

Reply:
I don't want to actually do anything with the command though. There's no response to send on our side when this completes, it's a 'silent no-op'?

Reply:
OK, this is because the "cmd" is actually a notification, which doesn't want a response.  It does, however, want to be freed!

I'll submit a followup PR which cleans this up everywhere...